cogent/Makefile

```diff
@@ -50,33 +41,49 @@ build: setup-config
 	$(E) "Building..."
 	$(CABAL) build
 
-configure: install-deps
+configure:
 	$(E) "Configuring"
 	$(CABAL) configure
-	$(CABAL) build
 
 sandbox:
-	$(E) "Building in a sandbox"
+ifneq ($(wildcard .cabal-sandbox/),)
+	$(E) "Using Existing sandbox..."
+else
+	$(E) "Building in a sandbox..."
 	$(CABAL) sandbox init
 	$(CABAL) sandbox add-source ../isa-parser
+endif
 
-install-deps: sandbox
+.install-deps:
 	$(E) "Installing dependencies"
-	$(CABAL) install --only-dependencies --force-reinstalls
+	$(CABAL) install --only-dependencies
 
 doc:
-	$(E) "Docs"
+	$(E) "Building Cogent docs..."
+	$(Q) make -C doc
 
 clean:
```

Comment:
We'd like to have a clean without cleaning `cogent` dependencies. (just like the cached data on travis)